nbdime/_version.py

```diff
@@ -6,18 +6,19 @@
 
 VersionInfo = namedtuple("VersionInfo", ["major", "minor", "micro", "releaselevel", "serial"])
 
-_specifier_ = {"a": "alpha", "b": "beta", "rc": "candidate", "": "final"}
+_specifier_ = {"a": "alpha", "b": "beta", "rc": "candidate"}
 
 __version__ = "4.0.0"
 
 parser = re.compile(r"^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<micro>\d+)((?P<releaselevel>a|b|rc)(?P<serial>\d+))?$")
 
 parsed_version = parser.match(__version__)
 groups = parsed_version.groupdict()
+
 version_info = VersionInfo(
     int(groups["major"]),
     int(groups["minor"]),
     int(groups["micro"]),
-    _specifier_[groups.get("releaselevel", "")],
+    _specifier_.get(groups.get("releaselevel", ""), "final"),
```

Comment:
I'm not sure if this is the right choice: if someone makes a mistake and bumps to "beta" instead of "b", then it will default to final since it cannot parse it?

Reply:
Would you mind sharing what error you were seeing that this will fix?

Reply:
```suggestion
    _specifier_[groups.get("releaselevel") or ""],
```

Reply:
The error is that if not found `groups.get("releaselevel") is None`. Therefore, as `None` is not a key of `_specificer`, it raises an `KeyError`.

I'll go for a more advance solution than Vidar's proposal as the propose solution does not address the case of release level being `beta` for example.

Reply:
My point was that "beta" would be an invalid value that *should* throw an error. Somewhere at least.. probably not good for it to throw in packaged code though, so this change is fine. But yes, let's try to get a regression test for the future as well.

Reply:
For documentation, `beta` is a valid specifier: https://peps.python.org/pep-0440/#pre-release-spelling